core/lib/vm/src/tracers/traits.rs

```diff
@@ -88,10 +81,22 @@ pub enum TracerExecutionStatus {
     Stop(TracerExecutionStopReason),
 }
 
-/// Stop the vm execution if the tracer conditions are met
-pub trait ExecutionEndTracer<H: HistoryMode> {
-    // Returns whether the vm execution should stop.
-    fn should_stop_execution(&self) -> TracerExecutionStatus {
-        TracerExecutionStatus::Continue
+impl TracerExecutionStatus {
+    /// Chose the stricter ExecutionStatus
+    /// If one of the statuses is Abort, then the result is Abort
+    /// If one of the statuses is Finish, then the result is Finish
+    /// Otherwise, the result is Continue
+    pub fn stricter(&self, other: &Self) -> Self {
+        match (self, other) {
+            (Self::Continue, Self::Continue) => Self::Continue,
+            (Self::Stop(TracerExecutionStopReason::Abort(reason)), _)
+            | (_, Self::Stop(TracerExecutionStopReason::Abort(reason))) => {
+                Self::Stop(TracerExecutionStopReason::Abort(reason.clone()))
+            }
+            (Self::Stop(TracerExecutionStopReason::Finish), _)
+            | (_, Self::Stop(TracerExecutionStopReason::Finish)) => {
+                Self::Stop(TracerExecutionStopReason::Finish)
+            }
```

Comment:
nit: it would be cleaner if the comment and arms (and their order) were consistent.